source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -331,14 +429,31 @@ void RedisCluster::RedisDiscoverySession::onResponse(
     return;
   }
 
-  auto slots = std::make_unique<std::vector<ClusterSlot>>();
-
+  auto cluster_slots = std::make_shared<std::vector<ClusterSlot>>();
+
+  // https://redis.io/commands/cluster-slots
+  // CLUSTER SLOTS represents nested array of redis instances, like this:
+  //
+  // 1) 1) (integer) 0                                      <-- start slot range
+  //    2) (integer) 5460                                   <-- end slot range
+  //
+  //    3) 1) "127.0.0.1"                                   <-- primary slot IP ADDR(HOSTNAME)
+  //       2) (integer) 30001                               <-- primary slot PORT
+  //       3) "09dbe9720cda62f7865eabc5fd8857c5d2678366"
+  //
+  //    4) 1) "127.0.0.2"                                   <-- replica slot IP ADDR(HOSTNAME)
+  //       2) (integer) 30004                               <-- replica slot PORT
+  //       3) "821d8ca00d7ccf931ed3ffc7e3db0599d2271abf"
+  //
   // Loop through the cluster slot response and error checks for each field.
+  uint64_t hostname_resolution_required_cnt = 0;
```

Comment:
This seems like it would result in a lifetime issue? I'd expect this to no longer be valid when the lambdas access it

Perhaps this should be wrapper in a shared_ptr?

Reply:
good point. But it seemed to work fine with the existing test cases. I'll wrap it in a `shared_ptr` anyway.